### DISCUSSION TITLE
Set of Element IDs of new elements in a refined mesh

### DISCUSSION THREAD

**Discussion by ritamcj** [idaholab/moose#18944] [Q&A General] [answered]
Hi everyone,
I am trying to access the set of new element IDs in a refined mesh at each time step. Are there any existing functions in Moose that can help with this? I looked into the doxygen files such as Marker.h, ValueThresholdMarker, QuadraturePointMarker and ElementIDInterface but could not find a suitable option. Additionally, I am unable to understand the boolean variable "refine_set" in ValueThresholdMarker. I would appreciate any leads.
Warm regards,
Ritam

**Comment by GiudGiud** [comment, marked as answer]
Hi Ritam

Where do you want this value stores? A vectorposprocessor? An AuxVariable? Would not be too hard to write an auxkernel to place that in an auxvariable. 

refine_set is simply a boolean to check if the user asked for a refine. It's to avoid doing that check at every quadrature point, and doing it only once in the constructor

One thing to know is that elements can be renumbered. You may want to disable that in the Mesh block if you want the ids pre- and post- refinement of some elements to remain the same. 

Guillaume

**Reply by ritamcj** [reply]
Hi Guillaume,
Thank you for your suggestions and explanation of "refine_set". I want to read the number of new elements inside FiniteStrainUObasedCP in tensor mechanics as I want to assign properties (Fp etc.) to the quadrature points on those elements. So, the number would be an integer that would be used to locate the correct quadrature points.
Ritam

**Reply by GiudGiud** [reply]
ok so you need this inside the code.
To only loop on the newly refined elements, you could loop over all elements and examine their refinement level. If the parent is not active, then the children element are new. Maybe the JUST_REFINED refinement state will help, I'm not sure.
See https://libmesh.github.io/doxygen/classlibMesh_1_1Elem.html for documentation on the element
